src/server/protocol_client.cc

```diff
@@ -51,18 +53,35 @@ using absl::StrCat;
 namespace {
 
 #ifdef DFLY_USE_SSL
+
+void PrintSSLError() {
```

Comment:
You implemented this and the `SSL_CHECK` macro twice. Maybe to avoid duplication you can move that in a header `sslerror.h` (or any other name) ?

Reply:
done